### ISSUE TITLE
Jump to a child directory

### ISSUE THREAD

**Issue by dufferzafar** [gsamokovarov/jump#31]
autojump used to have a `jc` shortcut to only jump to children of the current directories.

In #8 it was suggested that we could use `/` to distinguish among two parents, but I feel that if I'm in the current directory, I shouldn't need to distinguish it!

Having a command that only looks into the children of current seems to make more sense.

What are your thoughts?

**Comment by gsamokovarov** [comment]
As a design choice, I don't want to expose any other shell helpers than `j`. The `jump` binary itself contains a bunch of helper functionalities, like database management, pins and the likes.

If you need `jc`, you can define it youself:

```bash
# For bash, similar for the other shells.
jc() {
  j $(basename pwd) $*
}
```

**Comment by dufferzafar** [comment]
@gsamokovarov I'm okay with having such a function.

But it doesn't work! 

I use Zsh and tried:

```
jc() {
  j $(basename `pwd`)/$*
}
```

but it seems to ignore whatever i enter in $*

So: `jc grap` in a directory takes me to a child that doesn't even have `grap` in its path.

**Comment by gsamokovarov** [comment]
This works for me on `zsh`:

```zsh
jc() {
  j $(basename $PWD) $@
}
```

**Comment by gsamokovarov** [comment]
If this is useful, I can write shell specific versions and add it the README.

**Comment by gsamokovarov** [comment]
It's actually looking for direct children of `interview-prep` only. Don't know what the semantics of jc are.

`jc graph` is `j interview-prep/graph` which is trying to match `interview-prep/graph` as a direct substring in the path... kind of. 

```
/home/dufferzafar/dev/interview-prep/topics/graphs
```

Won't even be in the search list as jump will only filter out `interview-prep/*` for the match.

**Comment by dufferzafar** [comment]
Oh. 

Is there any way that we could match other children (not necessarily direct)

**Comment by gsamokovarov** [comment]
Jump matches on the basename or the "expanded basename" when slaches (or spaces) are present in the search. The motivation for the basename matches is that the fuzzy-searching algorithm can easily match a short input on a long absolute directory name and that's not what I usually want.

The "extended basename" adds a few levels of the directory name that's allowed to matched. With the current approach, what you are looking for won't be doable.

I can introduce `jump dir ** any_sub_dir` or `jump dir ** any_sub_dir` that will match on the absolute pathname, maybe then `jc` will be easier to implement. I'll get back to you in a few days.

**Comment by dufferzafar** [comment]
Thanks for taking the time to work on this :)

**Comment by gsamokovarov** [comment]
@dufferzafar Can you try the latest master with the following `jc` script:

```zsh
jc() {
  j "$(basename $PWD)/*/$@"
}
```

**Comment by dufferzafar** [comment]
I don't have a working Go installation, so can't build from source. I just use the binaries you upload on the releases page.

**Comment by gsamokovarov** [comment]
Okay, I'll test it by myself a bit more and push a release in a week or so.

**Comment by gsamokovarov** [comment]
@dufferzafar this should be solved in [`0.22.0`](https://github.com/gsamokovarov/jump/releases/tag/v0.22.0), at least from my testing.
